django/middleware/csrf.py

```diff
@@ -63,21 +101,28 @@ def rotate_token(request):
     """
     request.META.update({
         "CSRF_COOKIE_USED": True,
-        "CSRF_COOKIE": _get_new_csrf_key(),
+        "CSRF_COOKIE": _get_new_csrf_token(),
     })
 
 
 def _sanitize_token(token):
     # Allow only alphanum
-    if len(token) > CSRF_KEY_LENGTH:
-        return _get_new_csrf_key()
-    token = re.sub('[^a-zA-Z0-9]+', '', force_text(token))
-    if token == "":
-        # In case the cookie has been truncated to nothing at some point.
-        return _get_new_csrf_key()
+    if len(token) != CSRF_TOKEN_LENGTH:
+        return None
+    if re.search('[^a-zA-Z0-9]', force_text(token)):
```

Comment:
I have always thought `_` is considered alphanumeric, which is wrong, but still, in Python 3 "alphanumeric" is in Unicode:
```python
In [4]: 'א'.isalnum()
Out[4]: True
```
So, no.